scrapy/downloadermiddlewares/retry.py

```diff
@@ -63,6 +63,9 @@ def process_exception(self, request, exception, spider):
     def _retry(self, request, reason, spider):
         retries = request.meta.get('retry_times', 0) + 1
 
+        if 'max_retry_times' in request.meta:
+            self.max_retry_times = request.meta['max_retry_times']
```

Comment:
This would override the middleware `max_retry_times` attribute for all subsequent retried requests, while `request.meta['max_retry_times']` should only affect this particular request.